server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -353,20 +368,52 @@ public void writeTo(StreamOutput out) throws IOException {
     }
 
     /**
-     * Sets the search query for this request.
-     *
-     * @see org.elasticsearch.index.query.QueryBuilders
+     * Sets the query for this request.
      */
-    public SearchSourceBuilder query(QueryBuilder query) {
-        this.queryBuilder = query;
+    public SearchSourceBuilder query(QueryBuilder queryBuilder) {
+        subSearchSourceBuilders = new ArrayList<>();
+        if (queryBuilder != null) {
+            subSearchSourceBuilders.add(new SubSearchSourceBuilder(queryBuilder));
+        }
         return this;
     }
 
     /**
-     * Gets the query for this request
+     * Gets the query for this request.
+     *
+     * @return This will return {@code null} if there are no
+     * sub searches, a single {@link QueryBuilder} if there is only
+     * one sub search, and a combined {@link BoolQueryBuilder} if
+     * there are multiple sub searches where each sub search becomes
+     * a query with a should clause.
      */
     public QueryBuilder query() {
-        return queryBuilder;
+        if (subSearchSourceBuilders.isEmpty()) {
+            return null;
+        } else if (subSearchSourceBuilders.size() == 1) {
+            return subSearchSourceBuilders.get(0).getQueryBuilder();
+        } else {
+            BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+            for (SubSearchSourceBuilder subSearchSourceBuilder : subSearchSourceBuilders) {
+                boolQueryBuilder.should(subSearchSourceBuilder.getQueryBuilder());
+            }
```

Comment:
I am a bit doubtful about many existing callers of this method, that now end up retrieving the linear combination if sub_searches are provided. I can't decide if that's good (because they now get the boolean query automatically) or bad (because we have not explicitly decided that the linear combination applies to each of these cases). 

I wonder if this method should throw whenever it's called when there is more than one query, and we should introduce a new method that builds the linear combination so we track where that is specifically needed. Would that complicate things?

Reply:
I think as a follow up we should remove the query getter/setter, and force all callers to use subsearches to delegate to them what should be appropriate behavior, but for now this is what we use in QueryPhase to get the aggregations query for ranking or get a query for linear combination when hybrid search is used.

Reply:
I would even be ok leaving the single getter, but I think it's important to be explicit about when you need the linear combination vs when you expect to have single query: gut feeling at least.